### PR TITLE
libcontainer: rootfs_linux: support overlayfs

### PR DESCRIPTION
**Pull request by runcom**
As the runtime-spec allows it, we want to be able to specify overlayfs
mounts with:

    {
        "destination": "/etc/pki",
        "type": "overlay",
        "source": "overlay",
        "options": [
            "lowerdir=/etc/pki:/home/amurdaca/go/src/github.com/opencontainers/runc/rootfs_fedora/etc/pki"
        ]
    },

This patch takes care of allowing overlayfs mounts. Both RO and RW
should be supported.

Signed-off-by: Antonio Murdaca <runcom@redhat.com>

### CODE REVIEW

**Comment by runcom** [comment]
@rhatdan @mrunalp @rhvgoyal PTAL

File: libcontainer/rootfs_linux.go [hunk 1]
```diff
@@ -219,6 +219,11 @@ func mountToRootfs(m *configs.Mount, rootfs, mountLabel string) error {
 			}
 		}
 		return nil
+	case "overlay":
+		if err := mountPropagate(m, rootfs, mountLabel); err != nil {
+			return err
+		}
+		return label.SetFileLabel(m.Destination, mountLabel)
```
Comment (rhatdan) [inline libcontainer/rootfs_linux.go]: You should not call this.
The label would have to be specified in the overaly stuff above I believe.

Comment (runcom) [inline libcontainer/rootfs_linux.go]: removed

**Comment by mrunalp** [comment]
Maybe we could move this to the default section so it could work for most other options. @crosbymichael WDYT?

**Comment by crosbymichael** [comment]
Ya, probably better to do this in the default case and clean up anything else that does not require special handling.

**Comment by runcom** [comment]
alright, the default case should work just fine for overlay then. If we need other stuff I'll add them.

**Comment by crosbymichael** [comment]
@runcom i guess we mean, do we want to move the validation steps inside default?

**Comment by runcom** [comment]
@crosbymichael well sure, I wasn't fully sure those were needed for other mounts other than binds, I'll reopen.

**Comment by runcom** [comment]
can't reopen

**Comment by crosbymichael** [comment]
you have to restore your branch

**Comment by runcom** [comment]
yeah, I did another PR though (deeply sorry...) https://github.com/opencontainers/runc/pull/1314

**Pull request closed without merging**